ts/Series/TemperatureMap/TemperatureMapSeries.ts

```diff
@@ -0,0 +1,447 @@
+/* *
+ *
+ *  (c) 2010-2022 Rafal Sebestjanski
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *  Imports
+ *
+ * */
+
+import type TemperatureMapSeriesOptions from './TemperatureMapSeriesOptions';
+
+import Color from '../../Core/Color/Color.js';
+import ColorType from '../../Core/Color/ColorType';
+import MapBubbleSeries from '../MapBubble/MapBubbleSeries.js';
+import SeriesRegistry from '../../Core/Series/SeriesRegistry.js';
+import { SymbolKey } from '../../Core/Renderer/SVG/SymbolType';
+import TemperatureMapPoint from './TemperatureMapPoint.js';
+import U from '../../Core/Utilities.js';
+
+const {
+    correctFloat,
+    defined,
+    extend,
+    isArray,
+    isNumber,
+    merge,
+    pick
+} = U;
+
+/**
+ * The temperaturemap series type
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.temperaturemap
+ *
+ * @augments Highcharts.Series
+ */
+class TemperatureMapSeries extends MapBubbleSeries {
+
+    /* *
+     *
+     * Static properties
+     *
+     * */
+
+    /**
+     * A temperature series is a special case of the bubble series where
+     * multiple bubbles with different colors make one point. Multiple points
+     * can represent a temperature map or e.g. density.
+     *
+     * @sample maps/demo/temperaturemap
+     *         Temperature map chart
+     *
+     * @extends      plotOptions.mapbubble
+     * @product      highmaps
+     * @excluding    allowPointSelect, lineColor, lineWidth
+     * @optionparent plotOptions.temperaturemap
+     */
+    public static defaultOptions: TemperatureMapSeriesOptions = merge(
+        MapBubbleSeries.defaultOptions,
+        {
+            marker: {
+                lineWidth: 0
+            },
+            maxSize: 100,
+            minSize: 20,
+            opacity: 0.75,
+            /**
+             * States for a single point marker.
+             *
+             * @excluding hover, select
+             */
+            states: {
+                hover: {
+                    enabled: false
+                }
+            },
+            /**
+             * Set of colors for each point. Can be an array of strings (if 3
+             * colors defined, every next marker is 33.3% smaller) or an array
+             * of color stops (e.g. for `[[0.2, '#ff0000'], [1, '#0000ff']]`
+             * the smallest top marker is 5 times smaller than the biggest
+             * bottom one).
+             *
+             * @sample maps/plotoptions/temperaturemap-themes
+             *         Temperature colors themes
+             * @sample maps/plotoptions/temperaturemap-color-steps
+             *         Temperature colors with steps
+             *
+             * @default ["#ff0000", "#ffff00", "#00ff00", "#00ffff", "#0000ff"]
+             * @type {Array<ColorType>}
```

Comment:
The second type (probably `Array<[number,ColorType]>`) is not included in the API docs.